crates/primitives/src/bits/serde.rs

```diff
@@ -18,58 +18,82 @@ impl<const N: usize> Serialize for FixedBytes<N> {
 
 impl<'de, const N: usize> Deserialize<'de> for FixedBytes<N> {
     fn deserialize<D: Deserializer<'de>>(deserializer: D) -> Result<Self, D::Error> {
-        struct FixedVisitor<const N: usize>;
+        if deserializer.is_human_readable() {
```

Comment:
couldn't this be simplified to avoid the second visitor definition?

```
if deserializer.is_human_readable() {
    deserializer.deserialize_any(FixedVisitor::<N>)
} else {
    deserializer.deserialize_bytes(FixedVisitor::<N>)
```

Reply:
Very good point. I'll doublecheck to see if anything weird happens with those then unused methods, but I don't suspect it.

Reply:
changed